Practical Example/RW_interview_prep2/RW_interview_prep/Model/NetworkFetchRequest.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  FetchRequest.swift
+//  RW-takeHomeProject
+//
+//  Created by Duc Dang on 8/25/20.
+//  Copyright © 2020 Duc Dang. All rights reserved.
+//
+
+import Foundation
+
+class NetworkFetchRequest: ObservableObject {
+    
+    @Published var AllData = [ArticleAndVideoData]()
+    @Published var videoData = [ArticleAndVideoData]()
+    
+    
+    init() {
+        fetchArticleData()
+        fetchVideoData()
+    }
+
+    func fetchArticleData() {
+
+        if let url = URL(string: "https://api.jsonbin.io/b/5ed679357741ef56a566a67f") {
+            let session = URLSession(configuration: .default)
+            let task = session.dataTask(with: url) { (data, response, error) in
+                if error == nil {
+                    let decoder = JSONDecoder()
+                    if let safeData = data {
+                        do {
+                            let results = try decoder.decode(Arcdata.self, from: safeData)
+
+                            DispatchQueue.main.async {
+                                self.AllData.append(contentsOf: results.data)
+                            }
+                        } catch {
+                            print("error: ", error)
+                        }
+                    }
+                }
+            }
+            task.resume()
```

Comment:
This general bit of code is repeated multiple times in your project. I would move this to a `NetworkService` or `Interactor` that would have a function `fetch<T: Decodable>(from url: URL, completion: @escaping (T? -> Void))`. I.e., a generic function that will initiate a network request and try to decode the model.

Since you're already using SwiftUI and Combine, it makes sense to also fetch data with combine. Take a look here: https://developer.apple.com/documentation/foundation/urlsession/processing_url_session_data_task_results_with_combine